content/en/hosting/4.x/upgrade-troubleshooting.md

```diff
@@ -0,0 +1,160 @@
+
+## CHT 4.0.x - 4.3.x: CouchDB Crashes
+
+**[issue](https://github.com/medic/cht-core/issues/9286)**:  Starting an upgrade that involves view indexing can cause CouchDB to crash on large databases (>30m docs)
+
+HAProxy:
+
+```shell
+<150>Jul 26 20:57:39 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+<150>Jul 26 20:57:40 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+<150>Jul 26 20:57:41 haproxy[12]: 172.22.0.4,<NOSRV>,503,0,0,0,GET,/,-,medic,'-',217,-1,-,'-'
+```
+
+CouchDB
+
+```shell
+[notice] 2024-07-26T20:52:45.229027Z couchdb@127.0.0.1 <0.10998.4> dca1387a05 haproxy:5984 172.22.0.4 medic GET /_active_tasks 200 ok 1
+[notice] 2024-07-26T20:52:45.234397Z couchdb@127.0.0.1 <0.10986.4> 2715cd1e47 haproxy:5984 172.22.0.4 medic GET /medic-logs/_all_docs?descending=true&include_docs=true&startkey=%22upgrade_log%3A1722027165223%3A%22&limit=1 200 ok 6
+[notice] 2024-07-26T20:52:45.468469Z couchdb@127.0.0.1 <0.11029.4> 3bf85c6071 haproxy:5984 172.22.0.4 medic GET /_active_tasks 200 ok 1
+```
+
+**Fix:** 
+
+> Eventually I just abandoned using the upgrade API and started indexing process manually (by navigating to the views one by one). This seems to not trigger the same behavior.
+>
+> I'm suspecting we're ending up queuing too many requests because of the 2 minute timeout we set in API: https://github.com/medic/cht-core/blob/4.2.x/api/src/services/setup/view-indexer.js#L69
+> I believe this timeout doesn't end up cancelling the request at CouchDb level, so it's possible we're running out of file descriptors or something (???). Because there are no logs, it's not possible to know.
+ 
+
+## CHT 4.0.1 - 4.9.0: CouchDB restart causes all services to go down
+
+**Note** - This is a Docker only issue.
+
+**[issue](https://github.com/medic/cht-core/issues/9284)**:   A couchdb restart in single node docker takes down the whole instance.
+
+Haproxy continuously reports NOSRV errors like:
+
+```shell
+<150>Jul 25 18:11:03 haproxy[12]: 172.18.0.9,<NOSRV>,503,0,1001,0,GET,/,-,admin,'-',241,-1,-,'-'
+```
+
+API logs:
+
+```shell
+StatusCodeError: 503 - {"error":"503 Service Unavailable","reason":"No server is available to handle this request","server":"haproxy"}
+```
+
+nginx reports:
+
+```shell
+2024/07/25 18:40:28 [error] 43#43: *5757 connect() failed (111: Connection refused) while connecting to upstream, client: 172.18.0.1, 
+```
+
+
+**Fix:** Restart all services
+
+
+## CHT 4.2.4 - 4.3.x: CouchDB Crashes
+
+**[issue](https://github.com/medic/cht-core/issues/9617):** Starting an upgrade that involves view indexing can become stuck after indexing is finished
+
+upgrade process stalls after view indexes are built
+
+**Fix:**
+
+Unfortunately, the workaround is manual and very technical and involves:
+
+* identifying the upgrade log document in the medic-logs db.
```

Comment:
I have shared a new less technical workaround for this on the issue: https://github.com/medic/cht-core/issues/9617#issuecomment-2465299251

1. When API goes stuck after view indexing, simply restart API.
2. The admin upgrade page will say that the upgrade was interrupted, click retry upgrade.
3. Depending on the state of the database, you might see view indexing again. Depending on how many docs need to be indexed, indexing might get stuck again. Go back to 1 if that happens.
4. Eventually, when indexing jobs are short enough not to trigger a request hang, you will get the button to complete the upgrade.

Reply:
thanks! should be fixed in a745faab